Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -627,7 +718,7 @@ uint64_t SourceBuffer::maximumBufferSize() const
     const float bufferBudgetPercentageForVideo = .95;
     const float bufferBudgetPercentageForAudio = .05;
 
-    size_t maximum = settings().maximumSourceBufferSize();
+    size_t maximum = scriptExecutionContext()->settingsValues().maximumSourceBufferSize;
```

Comment:
Are we sure scriptExecutionContext() is always non nullptr?
If not, add a null check or store it in sourceBuffer constructor